treeherder/log_parser/parsers.py

```diff
@@ -39,227 +38,6 @@ def get_artifact(self):
-                # since we have no way of finding out the step metadata. This case occurs
-                # for the Taskcluster logs where content can fall between step markers.
-                self.start_step(lineno)
-            # Parse the line for errors, which if found, will be associated with the current step.
-            self.sub_parser.parse_line(line, lineno)
-            return
-
-        # This is either a "step started" or "step finished" marker line, eg:
-        # ========= Started foo (results: 0, elapsed: 0 secs) (at 2015-08-17 02:33:56.353866) =========
-        # ========= Finished foo (results: 0, elapsed: 0 secs) (at 2015-08-17 02:33:56.354301) =========
-
-        if step_marker_match.group('marker_type') == 'Started':
-            if self.state == self.STATES['step_in_progress']:
-                # We're partway through a step (ie: haven't seen a "step finished" marker line),
-                # but have now reached the "step started" marker for the next step. Before we
-                # can start the new step, we have to clean up the previous one - albeit using
-                # generic step metadata, since there was no "step finished" marker. This occurs
-                # in Taskcluster's logs when content falls between the step marker lines.
-                self.end_step(lineno)
-            # Start a new step using the extracted step metadata.
-            self.start_step(
-                lineno,
-                name=step_marker_match.group('name'),
-                timestamp=step_marker_match.group('timestamp'),
-            )
-            return
-
-        # This is a "step finished" marker line.
-
-        if self.state != self.STATES['step_in_progress']:
-            # We're not in the middle of a step, so can't finish one. Just ignore the marker line.
-            return
-
-        # Close out the current step using the extracted step metadata.
-        self.end_step(
-            lineno,
-            timestamp=step_marker_match.group('timestamp'),
-            result_code=int(step_marker_match.group('result_code')),
-        )
-
-    def start_step(self, lineno, name="Unnamed step", timestamp=None):
-        """Create a new step and update the state to reflect we're now in the middle of a step."""
-        self.state = self.STATES['step_in_progress']
-        self.stepnum += 1
-        self.steps.append(
-            {"name": name, "started": timestamp, "started_linenumber": lineno, "errors": [],}
-        )
-
-    def end_step(self, lineno, timestamp=None, result_code=None):
-        """Fill in the current step's summary and update the state to show the current step has ended."""
-        self.state = self.STATES['step_finished']
-        step_errors = self.sub_parser.get_artifact()
-        step_error_count = len(step_errors)
-        if step_error_count > settings.PARSER_MAX_STEP_ERROR_LINES:
-            step_errors = step_errors[: settings.PARSER_MAX_STEP_ERROR_LINES]
```

Comment:
@camd One thing I noticed from the failing tests I had commented out is that by removing the StepParser, we're no longer putting a cap (set to 100) on the number of failure lines per log we store (in general, we aren't doing any truncating of failure lines in this part of the log parsing that I can see). If this something we'd want to keep in the log parser? I wonder what implications it'd have for bug suggestions though, if any. How likely is 100+ failure lines for one live_backing log?

Reply:
Let's not put a limit in right now.  I don't know how often we hit that limit (if ever).  So let's not fix it unless it breaks.